node/pkg/driver/node.go

```diff
@@ -409,11 +410,21 @@ func (d *NodeService) mountFileSystemVolume(mpathDevice string, targetPath strin
 	targetPathWithHostPrefix := d.NodeUtils.GetPodPath(targetPath)
 	if !isTargetPathExists {
 		logger.Debugf("Target path directory does not exist. Creating : {%v}", targetPathWithHostPrefix)
-		err := d.Mounter.MakeDir(targetPathWithHostPrefix)
+		err := d.NodeUtils.MakeDir(targetPathWithHostPrefix)
 		if err != nil {
 			return status.Errorf(codes.Internal, "Could not create directory %q: %v", targetPathWithHostPrefix, err)
 		}
 	}
+
+	existingFormat, err := d.Mounter.GetDiskFormat(mpathDevice)
+	if err != nil {
+		logger.Errorf("Could not determine if disk {%v} is formatted, error: %v", mpathDevice, err)
+		return err
+	}
+	if existingFormat == "" {
+		d.NodeUtils.FormatDevice(mpathDevice, fsType)
+	}
+
 	logger.Debugf("Mount the device with fs_type = {%v} (Create filesystem if needed)", fsType)
 	return d.Mounter.FormatAndMount(mpathDevice, targetPath, fsType, nil) // Passing without /host because k8s mounter uses mount\mkfs\fsck
```

Comment:
are we going by assumption that because we formatted `FormatAndMount` wont format again?

Reply:
yes. same assumption we always had for reattach, import volume, etc...